src/api/worker/rest/CustomCacheHandler.ts

```diff
@@ -0,0 +1,126 @@
+import {ListElementEntity} from "../../common/EntityTypes.js"
+import {CalendarEvent, CalendarEventTypeRef} from "../../entities/tutanota/TypeRefs.js"
+import {TypeRef} from "@tutao/tutanota-utils"
+import {CUSTOM_MAX_ID, CUSTOM_MIN_ID, firstBiggerThanSecond, getElementId, LOAD_MULTIPLE_LIMIT} from "../../common/utils/EntityUtils.js"
+import {resolveTypeReference} from "../../common/EntityFunctions.js"
+import {getTypeId} from "@tutao/tutanota-utils/dist/TypeRef.js"
+import {ExposedCacheStorage} from "./EntityRestCache.js"
+import {EntityRestClient} from "./EntityRestClient.js"
+import {LateInitializedCacheStorage} from "./CacheStorageProxy.js"
+
+/**
+ * update when implementing custom cache handlers.
+ * add new types to the union when implementing new
+ * custom cache handlers.
+ */
+type CustomCacheHandledType = never | CalendarEvent
```

Comment:
I don't know if it makes sense to hardcode them all here

Reply:
We can just say that any list element can be handled here

Reply:
we can only handle list elements we have a customcachehandler impl for, but since you can't pass a type that you can't create a custom handler instance for to the custom handler map anyway this may be redundant

Reply:
turns out it's actually needed to make this watertight.

Reply:
I just think it's backwards to specify all the types in the general thing. If we want it we can make it generic parameter for the handler but it's probably not that important.